src/node_buffer.cc

```diff
@@ -588,8 +593,11 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
 
   // Then coerce everything that's not a string.
   if (!args[1]->IsString()) {
-    int value = args[1]->Uint32Value() & 255;
-    memset(ts_obj_data + start, value, fill_length);
+    uint32_t val;
+    if (args[1]->Uint32Value(ctx).To(&val)) {
```

Comment:
Fwiw, I still find it easier to read when using the return-on-error style rather than an `if (success) { … }` block

Reply:
Me too. Fixed.